compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -906,8 +906,14 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                                         // different from the received one
                                         // So we avoid suggestion method with Box<Self>
                                         // for instance
-                                        self.tcx.at(span).type_of(*def_id) != rcvr_ty
-                                            && self.tcx.at(span).type_of(*def_id) != rcvr_ty
```

Comment:
This is sort of unrelated to adding `EarlyBinder`, but I was confused by this... what's going on here?

Reply:
yeah this is really odd lol I have no idea why its checking the same condition twice...

Reply:
Hi, so I did a little digging on this and tracked it down to https://github.com/rust-lang/rust/pull/104422 (which renamed `actual` to `rcvr_ty`, whereas originally both names were in scope). I might try to think of a test case where the original check is needed